src/pluto/components/expected_value_differences_report.py

```diff
@@ -11,69 +50,89 @@ def __init__(self, data, v1, v2):
-            v2_values = self.values_by_field(self.v2_expected_records, field)
-
-            in1not2 = self.value_differences(v1_values, v2_values)
-            in2not1 = self.value_differences(v2_values, v1_values)
+        for comparison_name in FIELDS_OF_INTEREST.index:
+            st.markdown(f"#### Value differences for `{comparison_name}`")
+            
+            in1not2, in2not1 = self.value_differences_across_versions(comparison_name)
+            value_differences = pd.DataFrame.from_dict(
+                dict(
+                    [
+                        (f"in {self.v1} but not {self.v2}", [in1not2]),
+                        (f"in {self.v2} but not {self.v1}", [in2not1]),
+                    ]
+                ),
+                orient="index",
+                columns=["values"],
+            )
 
             if len(in1not2) == 0 and len(in2not1) == 0:
-                pass
+                st.write("None")
             else:
-                st.markdown(f"### Expected value difference for {field}")
-                if len(in1not2) != 0:
-                    st.markdown(f"* in {self.v1} but not in {self.v2}:")
-                    st.write(in1not2)
-                if len(in2not1) != 0:
-                    st.markdown(f"* in {self.v2} but not in {self.v1}:")
-                    st.write(in2not1)
+                st.dataframe(
+                    value_differences,
+                    column_config={
+                        "relevant fields": st.column_config.ListColumn(
+                            width="large",
+                        ),
+                    },
+                )
 
     @property
-    def expected_records(self):
+    def expected_records(self) -> dict:
         return self.df[self.df["v"].isin([self.v1, self.v2])].to_dict("records")
 
     @property
-    def v1_expected_records(self):
+    def v1_expected_records(self) -> list:
         return self.expected_records_by_version(self.v1)
 
     @property
-    def v2_expected_records(self):
+    def v2_expected_records(self) -> list:
         return self.expected_records_by_version(self.v2)
 
-    def expected_records_by_version(self, version):
+    def expected_records_by_version(self, version) -> list:
         return [i["expected"] for i in self.expected_records if i["v"] == version][0]
 
-    @property
-    def fields(self):
-        return [
-            "zonedist1",
-            "zonedist2",
-            "zonedist3",
-            "zonedist4",
-            "overlay1",
-            "overlay2",
-            "spdist1",
-            "spdist2",
-            "spdist3",
-            "ext",
-            "proxcode",
-            "irrlotcode",
-            "lottype",
-            "bsmtcode",
-            "bldgclasslanduse",
-        ]
-
     def values_by_field(self, df, field):
         return [i["values"] for i in df if i["field"] == field][0]
 
+    def values_by_fields(self, df: pd.DataFrame, fields: list):
+        values = []
```

Comment:
maybe
```suggestion
        return [self.values_by_field(df, field) for f in fields]
```

Reply:
sorry for the lack of type hints. `values_by_field()` returns a list, so this change would construct a list of lists, rather than a list of strings as intended by:

```python
values = values + self.values_by_field(df, field)
```

I added type hints to hopefully make this make more sense lol maybe there's still a list comprehension approach that's worth trying though?